openml-lightgbm/lightgbm-provider/src/main/java/com/feedzai/openml/provider/lightgbm/LightGBMBinaryClassificationModel.java

```diff
@@ -109,6 +110,18 @@ public void close() throws Exception {
      * @return the double [ ]
      */
     public double[] getFeatureContributions(final Instance instance) {
+        return getFeatureContributions(instance, new LightGBMExplanationsConfig());
+    }
+
+    /**
+     * Get features contributions double [ ].
+     *
+     * @param instance the instance
+     * @param explanationsConfig TODO
+     * @return the double [ ]
+     * @since @@@feedzai.next.release@@@
+     */
+    public double[] getFeatureContributions(final Instance instance, final LightGBMExplanationsConfig explanationsConfig) {
```

Comment:
IMO I would drop this.

Afaik this is not a necessity for the moment (is not even being considered for the future). besides that having a parameter specific of the model would mess with the API. that would not be generic anymore. so to avoid that I think that is better to simple have the method above

Reply:
Just to complement, basically FdzRF requires a class index so we can get all feature contributions for the specified class. However SHAP explanation doesn't require a class index.

For this reason, we need to add a new parameter so the FdzRF#getFeatureContributions can predict contributions.
Since it's required, the config object can be also used in the future, if Microsoft provides us the api to choose other explanations.

Reply:
This has been redesign, please take a new look